server/src/main/java/org/opensearch/common/blobstore/VerifyingMultiStreamBlobContainer.java

```diff
@@ -31,4 +32,17 @@ public interface VerifyingMultiStreamBlobContainer extends BlobContainer {
      * @throws IOException if any of the input streams could not be read, or the target blob could not be written to
      */
     void asyncBlobUpload(WriteContext writeContext, ActionListener<Void> completionListener) throws IOException;
+
+    /**
+     * Creates and populates a list of {@link java.io.InputStream} from the blob stored within the repository, returned
+     * within an async callback using the {@link ReadContext} object.
+     * Defaults to using multiple streams, when feasible, unless a single stream is forced using @param forceSingleStream.
+     * An {@link IOException} is thrown if requesting any of the input streams fails, or reading metadata for the
+     * requested blob fails
+     * @param blobName          Name of the blob to be read using the async mechanism
+     * @param forceSingleStream Value to denote forced use of a single stream within the returned object
+     * @param listener  Async listener for {@link ReadContext} object which serves the input streams and other metadata for the blob
+     * @throws IOException if any of the input streams could not be requested, or reading metadata for requested blob fails
+     */
+    void asyncBlobDownload(String blobName, boolean forceSingleStream, ActionListener<ReadContext> listener) throws IOException;
```

Comment:
we can skip `forceSingleStream` . Based on `size` , the user can determine if they need to call multipart download or not .

Reply:
I think that information will be repository plugin specific. In cases where the user wants to use a single stream with `async` stream support, this will help.

Reply:
I am in favor of not providing flexibility which doesn't add much value. Do we  see use case for this option ?

Reply:
I got rid of it and instead updated the design. Please have a look at - https://github.com/opensearch-project/OpenSearch/pull/9182#issuecomment-1679169401